### DISCUSSION TITLE
Roadmapping: Support and/or documentation for debugging plugins

### DISCUSSION THREAD

**Discussion by robgjansen** [shadow/shadow#2035] [Show and tell]
- What do we do when program runs into an error case? e.g.:
    - program exits cleanly but returns nonzero error code
    - shadow deadlocks
    - program segfauls
- We have docs explaining how to attach GDB, which helps debug shadow, but figuring out how to debug a given managed process is more complicated but not documented.

Plan for v2.1:
  - New cli-only option called something like "--debug-host" that accepts a comma-separated list of hostnames from the config.yaml file. Whenever any of those hosts forks a process, Shadow will pause in between fork and exec of that process (we'll work out the details of who gets paused - either the child or shadow - later).
  - Shadow will have to print out the Linux PID of the child so the user can attach to it with GDB, or continue it if the user doesn't want to attach to that process.
  - We should document how this works in the dev part of the guide.
  - Useful for development so we'll try to do for release v2.1

notes:
    - 'handle SIGSYS noprint nostop'
    - 'handle SIGSEGV noprint nostop' -- probably won't need this one

**Comment by stevenengler** [comment]
This is tracked in https://github.com/shadow/shadow/issues/1873, with a list of objectives in https://github.com/shadow/shadow/issues/1873#issuecomment-1088944664.
